### PR TITLE
first draft of thresher; processing yield and fees

### PR DESCRIPTION
**Pull request by jorijnsmit**
solution to #56

### CODE REVIEW

**Comment by jorijnsmit** [comment]
ideally we wait for #28, #131 and badger tree emission redirection before editing and running this.

File: scripts/badger/thresher.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        expected * SLIPPAGE
+    )
+
+    # deposit sbtc_lp directly into ibbtc_lp
+    SAFE.curve.deposit(IBBTC_LP, [0, SBTC_LP.balanceOf(SAFE) * DUSTY])
+
+    # finally dogfood all into our own ibbtc sett
+    IBBTC_LP.approve(BIBBTC, 2**256-1)
+    BIBBTC.depositAll()
+    IBBTC_LP.approve(BIBBTC, 0)
```
Comment (petrovska-petro) [inline scripts/badger/thresher.py:137]: I think this section could be done less dusty relying into this [zapper](https://github.com/Badger-Finance/badger-multisig/blob/main/helpers/addresses.py#L436) instead

The following [deposit](https://etherscan.io/address/0xc5b45ab3d237b0dfc8f8808df39d49ebde8171fe#code#L1364)  method would be handy

Comment (petrovska-petro) [reply scripts/badger/thresher.py:137]: looking into at it more, i think it will leave hanging the sbtc_lp balance if going the other route...

**Review by petrovska-petro — APPROVED** [review]
- Consolidating into key sett tokens: bvecvx, bcvxcrv & bibbtc ✅ 

- Unwraps curve, sushi & uni ✅ 

- Consolidate usdc, dai and usdt into 3pool ✅ 


Snap:

```
snapshot result for 0x042B32Ac6b453485e357938bdC38e0340d4b9276:
                              balance_before               balance_after                balance_delta
symbol                                                                                               
BADGER              5,000.000000000000000000   20,462.657579641411777907    15,462.657579641411777907
ibBTC                   0.038950617857484599        0.001331763539982798        -0.037618854317501801
DIGG                    0.078975719000000000        0.501798870000000000         0.422823151000000000
USDT                2,946.129307000000000000        0.000000000000000000    -2,946.129307000000000000
USDC              419,579.019201000000000000        0.000000000000000000  -419,579.019201000000000000
DAI               111,450.530166875737873018        0.000000000000000000  -111,450.530166875737873018
CRV                   931.548895506701708221        0.000000000000000000      -931.548895506701708221
WBTC                    0.601531700000000000        0.028248580000000000        -0.573283120000000000
renBTC                  0.139954720000000000        0.000699780000000000        -0.139254940000000000
WETH                   41.913058624872964898       91.653546680442072357        49.740488055569107459
bBADGER                98.049355725222614268        0.000000000000000000       -98.049355725222614268
crvRenWBTC              0.227771066825237180        0.000000000000000000        -0.227771066825237180
crvRenWSBTC             0.044418098780758964        0.001183824719320202        -0.043234274061438762
tbtc/sbtcCrv            0.119959810768566656        0.000000000000000000        -0.119959810768566656
SLP                     0.000000039313031053        0.000000000000000000        -0.000000039313031053
UNI-V2                  0.000000000316226766        0.000000000000000000        -0.000000000316226766
SLP                     0.000002256577742932        0.000000000000000000        -0.000002256577742932
SLP                   752.198636734524115950        0.000000000000000000      -752.198636734524115950
hCRV                    0.138509454978643360        0.000000000000000000        -0.138509454978643360
bBTC/sbtcCRV            0.009152098801314979        0.000000000000000000        -0.009152098801314979
oBTC/sbtcCRV            0.046465433934354089        0.000000000000000000        -0.046465433934354089
pBTC/sbtcCRV            0.002035191408957473        0.000000000000000000        -0.002035191408957473
ibbtc/sbtcCRV-f         0.060076571242109943        0.000000000000000000        -0.060076571242109943
crv3crypto              0.323173563701192738        0.000000000000000000        -0.323173563701192738
3Crv                    0.000000000000000000  523,217.869244564952582499   523,217.869244564952582499
MIM-3LP3CRV-f           0.039827993552528545        0.000000000000000000        -0.039827993552528545
FRAX3CRV-f              0.039706063903846634        0.000000000000000000        -0.039706063903846634
CVX                   390.337135617178279022        0.000000000000000000      -390.337135617178279022
cvxCRV              9,129.027440447304847181        0.000000000000000000    -9,129.027440447304847181
bcvxCRV            20,254.763318954187920209   27,601.671461081581751278     7,346.908142127393831069
bcrvRenWBTC             0.083705682871655912        0.000000000000000000        -0.083705682871655912
bveCVX              5,890.856337802452189800    6,281.193473419630468822       390.337135617178279022
bibbtc/sbtcCRV-f       22.244492276616633500       28.504683514044135314         6.260191237427501814
bcrv3crypto           104.411400561752669326        0.000000000000000000      -104.411400561752669326
```

**Review by jalbrekt85 — APPROVED** [review]
clever way to filter lp tokens from registry, token balances from snapshot look good

**Pull request merged**